### PR TITLE
Revised New vehicle data GearStatus

### PR DESCRIPTION
**Pull request by atiwari9**
This proposal is to add new vehicle data `GearStatus` to know drive mode and actual gear the transmission has selected.

### CODE REVIEW

**Comment by atiwari9** [comment]
@theresalech - Ready to be reviewed.

File: proposals/0266-New-vehicle-data-GearStatus.md [hunk 1]
```diff
@@ -28,69 +28,53 @@ We need to add `GearStatus` for `GetVehicleData`, `SubscribeVehicleData`, `Unsub
...
+		<description>Semi automatic transmission.</description>
 	</element>
-	<element name="SEVENTH">
+	<element name="DUAL_CLUTCH">
+		<description>Dual clutch transmission.</description>
 	</element>
-	<element name="EIGHTH">
-	</element>
-	<element name="NINTH">
+	<element name="CONTINUOSLY_VARIABLE">
```
[the remark this answers is not in the document]
Comment (atiwari9) [reply proposals/0266-New-vehicle-data-GearStatus.md]: Done

File: proposals/0266-New-vehicle-data-GearStatus.md [hunk 2]
```diff
@@ -28,69 +28,53 @@ We need to add `GearStatus` for `GetVehicleData`, `SubscribeVehicleData`, `Unsub
...
+		<description>Continuosly variable transmission(CVT).</description>
```
[the remark this answers is not in the document]
Comment (atiwari9) [reply proposals/0266-New-vehicle-data-GearStatus.md]: Done

File: proposals/0266-New-vehicle-data-GearStatus.md [hunk 3]
```diff
@@ -179,22 +168,57 @@ We need to add `GearStatus` for `GetVehicleData`, `SubscribeVehicleData`, `Unsub
...
+	<element name="AUTOMATIC">
+		<description>Automatic transmission.</description>
+	</element>
+	<element name="SEMI_AUTOMATIC">
+		<description>Semi automatic transmission.</description>
+	</element>
+	<element name="DUAL_CLUTCH">
+		<description>Dual clutch transmission.</description>
+	</element>
+	<element name="CONTINUOSLY_VARIABLE">
```
[the remark this answers is not in the document]
Comment (atiwari9) [reply proposals/0266-New-vehicle-data-GearStatus.md]: Done

File: proposals/0266-New-vehicle-data-GearStatus.md [hunk 4]
```diff
@@ -179,22 +168,57 @@ We need to add `GearStatus` for `GetVehicleData`, `SubscribeVehicleData`, `Unsub
...
+		<description>Continuosly variable transmission(CVT).</description>
```
[the remark this answers is not in the document]
Comment (atiwari9) [reply proposals/0266-New-vehicle-data-GearStatus.md]: Done

**Comment by atiwari9** [comment]
@theresalech 

>deprecate prndl vehicle data item

Yes
>add new vehicle data item gearStatus

Yes

>key selectedGear as part of GearStatus struct

Yes, but the name is `actualGear` as specified in original proposal and option 2.

I have also corrected the typos

**Comment by joeygrover** [comment]
>Yes, but the name is actualGear as specified in original proposal and option 2.

It is `selectedGear` and `actualGear` per the vote regarding [this comment](https://github.com/smartdevicelink/sdl_evolution/issues/881#issuecomment-564258669). You have it as `gearDriveMode ` and it should be `selectedGear` Please update.

**Comment by atiwari9** [comment]
@joeygrover - I have updated the proposal as per Option 2 in [this comment](https://github.com/smartdevicelink/sdl_evolution/issues/881#issuecomment-564139080) plus adding new param `transmissionType` in `GearStatus` struct as agreed upon during the meeting, More over, `gearDriveMode` makes more sense to represent _PRNDL/PRNDS_  rather than `selectedGear` which can easily be confused with `actualGear`(gear selected by transmission control module). We should keep it as it is, any concerns can be discussed during review.

**Comment by joeygrover** [comment]
Unfortunately that is not what was voted on, please update to what revisions were requested and voice your opinion on the next review. We are unable to make any choices or deviations from the vote in this forum. Since it's a relatively minor change it shouldn't need another round of review after and could be accepted with revisions.

**Comment by atiwari9** [comment]
@joeygrover - I guess the miss here is the understanding about what we were voting on, In the end i did specify that we are voting on Option 2 + Kujtim's suggestion to add transmissionType. It was clarified from PM side that "Yes as long as Option2 means deprecating `prndl` and adding `gearStatus`". So if that was the last item specified, that is what everyone voted on.

Since this is a small thing which i can address during review as well and i do want to honor the written comment in the review ticket, i'd make the change for `gearDriveMode` -> `selectedGear`. 

But i hope you understand the concern here, there is a gap between what we thought was voted upon vs what was actually documented. We need to find a way to avoid that for future before it happens for a bigger change than this one.

**Comment by joeygrover** [comment]
Your concern is noted, however that is not what the project maintainer has as what was voted on. Please update and voice your concern on the next review.

**Comment by atiwari9** [comment]
@theresalech  - Ready to be reviewed

**Pull request merged**